specification/communication/data-plane/PhoneNumbers/preview/2021-10-25-preview/phonenumbers.json

```diff
@@ -0,0 +1,1639 @@
+            "name": "PhoneNumberType",
+            "modelAsString": true
+          }
+        },
+        "assignmentType": {
+          "description": "The assignment type of the phone numbers to search for. A phone number can be assigned to a person, or to an application.",
+          "enum": [
+            "person",
+            "application"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberAssignmentType",
+            "modelAsString": true
+          }
+        },
+        "capabilities": {
+          "$ref": "#/definitions/PhoneNumberCapabilities"
+        },
+        "areaCode": {
+          "description": "The area code of the desired phone number, e.g. 425.",
+          "type": "string"
+        },
+        "quantity": {
+          "format": "int32",
+          "description": "The quantity of desired phone numbers. The default value is 1.",
+          "default": 1,
+          "maximum": 2147483647,
+          "minimum": 1,
+          "type": "integer"
+        }
+      }
+    },
+    "PhoneNumberCost": {
+      "description": "The incurred cost for a single phone number.",
+      "required": [
+        "amount",
+        "billingFrequency",
+        "currencyCode"
+      ],
+      "type": "object",
+      "properties": {
+        "amount": {
+          "format": "double",
+          "description": "The cost amount.",
+          "type": "number"
+        },
+        "currencyCode": {
+          "description": "The ISO 4217 currency code for the cost amount, e.g. USD.",
+          "type": "string"
+        },
+        "billingFrequency": {
+          "description": "The frequency with which the cost gets billed.",
+          "enum": [
+            "monthly"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BillingFrequency",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "PhoneNumberSearchResult": {
+      "description": "The result of a phone number search operation.",
+      "required": [
+        "assignmentType",
+        "capabilities",
+        "cost",
+        "phoneNumbers",
+        "phoneNumberType",
+        "searchExpiresBy",
+        "searchId"
+      ],
+      "type": "object",
+      "properties": {
+        "searchId": {
+          "description": "The search id.",
+          "type": "string"
+        },
+        "phoneNumbers": {
+          "description": "The phone numbers that are available. Can be fewer than the desired search quantity.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "phoneNumberType": {
+          "description": "The phone number's type, e.g. geographic, or tollFree.",
+          "enum": [
+            "geographic",
+            "tollFree"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberType",
+            "modelAsString": true
+          }
```

Comment:
Consider pulling all these repeated enums into definitions that you `$ref` so these can't ever get out of sync.